src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -719,19 +720,28 @@ def create_or_upgrade_app(
                         self.execute_post_deploy_hooks()
                         return
 
-                    except ProgrammingError as err:
-                        if err.errno == CANNOT_DISABLE_MANDATORY_TELEMETRY:
+                    except (UserInputError, ProgrammingError) as err:
```

Comment:
Not in love with this style of error handling, but unsure what would be a better approach without further refactoring